plugins/database/hana/hana_test.go

```diff
@@ -0,0 +1,172 @@
+package hana
+
+import (
+	"database/sql"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/vault/builtin/logical/database/dbplugin"
+	"github.com/hashicorp/vault/plugins/helper/database/connutil"
+)
+
+var (
+	testHANAImagePull sync.Once
+)
+
+func TestHANA_Initialize(t *testing.T) {
```

Comment:
I'm actually less concerned about the overall logic and more about actually testing it against a real server. It's a possibility, sure.

BTW, can you send me an email? It's my email address at my company dot com.